lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -610,34 +693,51 @@ namespace Js
         }
 
         return JavascriptString::NewCopySz(wgl->WindowsGetStringRawBuffer(*locale, NULL), scriptContext);
+#endif
     }
 
     Var IntlEngineInterfaceExtensionObject::EntryIntl_GetDefaultLocale(RecyclableObject* function, CallInfo callInfo, ...)
     {
         EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
 
-        WCHAR defaultLocale[LOCALE_NAME_MAX_LENGTH];
+        char16 defaultLocale[LOCALE_NAME_MAX_LENGTH];
         defaultLocale[0] = '\0';
 
-        if (GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0 || defaultLocale[0] == '\0')
+        if (
+#if defined(INTL_WINGLOB)
+            // TODO (doilij) investigate why the PAL definition of this function wasn't found in Linux build.
+            GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0
+#else
+            IcuIntlAdapter::GetUserDefaultLocaleName(defaultLocale, _countof(defaultLocale)) == 0
+#endif
+            || defaultLocale[0] == '\0')
         {
+#if defined(INTL_WINGLOB)
             return scriptContext->GetLibrary()->GetUndefined();
+#else
+            Output::Print(_u("EntryIntl_GetDefaultLocale > IcuIntlAdapter::GetUserDefaultLocaleName returned 0: returning null to fallback to JS\n"));
+
+            return scriptContext->GetLibrary()->GetNull();
```

Comment:
Yes. Undefined is when the answer should actually return undefined to the JS logic, null is a sentinel to tell that the JS logic should fall back on another method of determining a useful value (basically, API is not implemented yet in native, but JS can use some reasonable default or an alternate JS-only algorithm to approximate the answer).